File: python/download_td.py
from qcportal import PortalClient
from openff.qcsubmit.results import TorsionDriveResultCollection

ds_cache = "../testfiles/download_td_want.json"
td_datasets = ["OpenFF multiplicity correction torsion drive data v1.1"]

client = PortalClient("https://api.qcarchive.molssi.org:443")
dataset = TorsionDriveResultCollection.from_server(
    client=client,
    datasets=td_datasets,
    spec_name="default",
)
with open(ds_cache, "w") as out:
    out.write(dataset.json(indent=2))

File: tests/download_td.rs
use std::fs::read_to_string;

use openff_qcsubmit::results::TorsionDriveResultCollection;
use qcportal::PortalClient;

#[test]
fn download_td() {
    let datasets =
        vec!["OpenFF multiplicity correction torsion drive data v1.1"];
    let client = PortalClient::new("https://api.qcarchive.molssi.org:443");
    let spec_name = "default";
    let dataset =
        TorsionDriveResultCollection::from_server(client, datasets, spec_name);
    let got = dataset.json(2);
    let want = read_to_string("testfiles/download_td_want.json").unwrap();
    // I'm not sure this order is deterministic, but it's passing for now.
    assert_eq!(got, want);
}
